rskj-core/src/main/java/org/ethereum/rpc/dto/BlockResultDTO.java

```diff
@@ -127,17 +129,12 @@ public static BlockResultDTO fromBlock(Block b, boolean fullTx, BlockStore block
 
         Coin mgp = b.getMinimumGasPrice();
 
-        List<Object> transactions = new ArrayList<>();
         List<Transaction> blockTransactions = b.getTransactionsList();
-        if (fullTx) {
-            for (int i = 0; i < blockTransactions.size(); i++) {
-                transactions.add(new TransactionResultDTO(b, i, blockTransactions.get(i)));
-            }
-        } else {
-            for (Transaction tx : blockTransactions) {
-                transactions.add(tx.getHash().toJsonString());
-            }
-        }
+        // For full tx will present as TransactionResultDTO otherwise just as transaction hash
+        List<Object> transactions = IntStream.range(0, blockTransactions.size())
+                .mapToObj(txIndex -> toTransactionResult(txIndex, b, fullTx, skipRemasc))
+                .filter(o -> o != null)
```

Comment:
I'd use `.filter(Objects::nonNull)` (I´ve seen it around the node and think it´s more legible=

Reply:
ohhh nice, i was trying to find that one!

Reply:
https://github.com/rsksmart/rskj/pull/1576/commits/ae3ff0624728affb4ea5983b6f8a358b98dc66aa